### PR TITLE
# update threejs ,addfunction and add outlinepass and add SMAAPass

### PR DESCRIPTION
**Pull request by FishOrBear**
update threejs types.

### CODE REVIEW

File: types/three/three-OutlinePass.d.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    Camera,
+    Color,
+    Face3,
+    Light,
+    Material,
+    Object3D,
+    Scene,
+    Vector2,
+    Vector3,
+    Vector4
```
Comment (efokschaner) [inline types/three/three-OutlinePass.d.ts]: mind removing the unused imports?

File: types/three/three-SMAAPass.d.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    Camera,
+    Color,
+    Face3,
+    Light,
+    Material,
+    Object3D,
+    Scene,
+    Vector2,
+    Vector3,
+    Vector4
```
Comment (efokschaner) [inline types/three/three-SMAAPass.d.ts]: mind removing the unused imports?

File: types/three/index.d.ts [hunk 3]
```diff
@@ -29,5 +29,7 @@ export * from "./three-trackballcontrols";
 export * from "./three-transformcontrols";
 export * from "./three-vrcontrols";
 export * from "./three-vreffect";
+export * from "./three-OutlinePass";
+export * from "./three-SMAAPass";
```
Comment (efokschaner) [inline types/three/index.d.ts]: Please keep these alphabetically ordered. On that note, they're almost entirely lowercase and we should match that too.

**Comment by efokschaner** [comment]
@FishOrBear Thanks for taking the time to PR these fixes, I left some comments to address, mostly style stuff.

File: types/three/three-OutlinePass.d.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+} from "./three-core";
+export class OutlinePass
+{
+    constructor(size: Vector2, scene: Scene, camer: Camera);
```
Comment (efokschaner) [inline types/three/three-OutlinePass.d.ts]: Mind including the rest of the args here?

File: types/three/three-OutlinePass.d.ts (hunk 1 above)
Comment (FishOrBear) [reply types/three/three-OutlinePass.d.ts]: ok

File: types/three/index.d.ts (hunk 3 above)
Comment (FishOrBear) [reply types/three/index.d.ts]: ok

File: types/three/three-OutlinePass.d.ts (hunk 4 above)
Comment (FishOrBear) [reply types/three/three-OutlinePass.d.ts]: ok

**Comment by efokschaner** [comment]
Thanks @FishOrBear , I'm :+1: on this.

**Pull request merged**